planning/behavior_velocity_planner/autoware_behavior_velocity_blind_spot_module/src/util.cpp

```diff
@@ -0,0 +1,314 @@
+  if (start == end) {
+    // there is only one point in the path
+    return std::nullopt;
+  }
+  for (size_t i = 0; i < p.points.size(); ++i) {
+    if (hasLaneIds(p.points.at(i), id)) {
+      if (!found) {
+        // found interval for the first time
+        found = true;
+        start = i;
+      }
+    } else if (found) {
+      // prior point was in the interval. interval ended
+      end = i;
+      break;
+    }
+  }
+  start = start > 0 ? start - 1 : 0;  // the idx of last point before the interval
+  return found ? std::make_optional(std::make_pair(start, end)) : std::nullopt;
+}
+}  // namespace
+
+std::optional<InterpolatedPathInfo> generateInterpolatedPathInfo(
+  const lanelet::Id lane_id, const tier4_planning_msgs::msg::PathWithLaneId & input_path,
+  rclcpp::Logger logger)
+{
+  constexpr double ds = 0.2;
+  InterpolatedPathInfo interpolated_path_info;
+  if (!splineInterpolate(input_path, ds, interpolated_path_info.path, logger)) {
+    return std::nullopt;
+  }
+  interpolated_path_info.ds = ds;
+  interpolated_path_info.lane_id = lane_id;
+  interpolated_path_info.lane_id_interval =
+    findLaneIdInterval(interpolated_path_info.path, lane_id);
+  return interpolated_path_info;
+}
+
+std::optional<lanelet::ConstLanelet> getSiblingStraightLanelet(
+  const lanelet::Lanelet assigned_lane,
+  const lanelet::routing::RoutingGraphConstPtr routing_graph_ptr)
+{
+  for (const auto & prev : routing_graph_ptr->previous(assigned_lane)) {
+    for (const auto & following : routing_graph_ptr->following(prev)) {
+      if (std::string(following.attributeOr("turn_direction", "else")) == "straight") {
+        return following;
+      }
+    }
+  }
+  return std::nullopt;
+}
+
+lanelet::ConstLanelet generateHalfLanelet(
+  const lanelet::ConstLanelet lanelet, const TurnDirection & turn_direction,
+  const double ignore_width_from_centerline)
+{
+  lanelet::Points3d lefts, rights;
+
+  const double offset = (turn_direction == TurnDirection::LEFT) ? ignore_width_from_centerline
+                                                                : -ignore_width_from_centerline;
+  const auto offset_centerline = lanelet::utils::getCenterlineWithOffset(lanelet, offset);
+
+  const auto original_left_bound =
+    (turn_direction == TurnDirection::LEFT) ? lanelet.leftBound() : offset_centerline;
+  const auto original_right_bound =
+    (turn_direction == TurnDirection::LEFT) ? offset_centerline : lanelet.rightBound();
+
+  for (const auto & pt : original_left_bound) {
+    lefts.emplace_back(pt);
+  }
+  for (const auto & pt : original_right_bound) {
+    rights.emplace_back(pt);
+  }
+  const auto left_bound = lanelet::LineString3d(lanelet::InvalId, std::move(lefts));
+  const auto right_bound = lanelet::LineString3d(lanelet::InvalId, std::move(rights));
+  auto half_lanelet = lanelet::Lanelet(lanelet::InvalId, left_bound, right_bound);
+  return half_lanelet;
+}
+
+lanelet::ConstLanelet generateExtendedAdjacentLanelet(
```

Comment:
- [X] same impl is as before
- [x] check planner_param_.adjacent_extend_width is passed as before